include/deal.II/matrix_free/evaluation_kernels.h

```diff
@@ -2067,20 +1942,58 @@ namespace internal
                                                                   values,
                                                                   gradients,
                                                                   do_values);
-            work_tangential<1, 2, false>(shape_data[1], values, values);
-            work_tangential<0, 2, false>(shape_data[1], values, values);
-            work_normal<2, false>(shape_data[0],
-                                  values,
-                                  values_dofs,
-                                  add_into_values_array);
+            tensor_product_apply_aniso_tangential<dim,
+                                                  fe_degree,
+                                                  n_q_points_1d,
+                                                  1,
+                                                  2,
+                                                  false>(shape_data[1],
+                                                         values,
+                                                         values);
+            tensor_product_apply_aniso_tangential<dim,
+                                                  fe_degree,
+                                                  n_q_points_1d,
+                                                  0,
+                                                  2,
+                                                  false>(shape_data[1],
+                                                         values,
+                                                         values);
```

Comment:
Would it be possible to refactor these repeated calls into a lambda to avoid having template arguments split over 6 lines repeatedly?

Reply:
@kronbichler What do you think about a class so that one could write something like:

```
            tensor_product_apply_aniso<dim, fe_degree, n_q_points, false>(shape_data);

            tensor_product_apply_aniso.tangential<1, 2>(values, values);
```

Reply:
Yes, I had been thinking about a similar solution to reduce the number of template arguments: All parameters that need to be changed at a specific call site get function templates, everything else class templates. I could not figure that out for lambdas because lambda template arguments seem to be [C++20](https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2017/p0428r2.pdf), and I would really profit from the template argument.

Reply:
@kronbichler How do you want to proceed? I don't mind the wrapped function calls.

Reply:
I will change this here. I plan a follow-up PR to clean up some more classes (with backward compatibility option).

Reply:
I now switched to classes, which allows me to fit the call in mostly one, sometimes up to 3 lines. Much better now.